src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -181,10 +184,20 @@ AsyncLogWriter* AsyncLogWriter::instance() {
   return _instance;
 }
 
-// write() acquires and releases _io_sem even _buffer is empty.
-// This guarantees all logging I/O of dequeued messages are done when it returns.
+// NOT MT-safe! see the comments in the header file.
 void AsyncLogWriter::flush() {
   if (_instance != nullptr) {
-    _instance->write();
+    {
+      using none = LogTagSetMapping<LogTag::__NO_TAG>;
+      AsyncLogLocker locker;
+      LogDecorations d(LogLevel::Off, none::tagset(), LogDecorators::None);
+      AsyncLogMessage token(nullptr, d, nullptr);
+
+      // not disposable
```

Comment:
I don't know what "not disposable" means.

Reply:
The reason I don't call enqueue_locked() here because it may be dropped due to buffer size. 
This is a control token. if it gets dropped, flush() will get stuck forever. 

not droppable?

Reply:
How about:
// Push directly in-case we are at logical max capacity, as this must not get dropped

I'm curious now about the fact the buffer size seems only a logical size constraint not an actual out-of-space constraint.

Reply:
currently, it's a soft limit.